client/packages/programs/src/JsonForms/common/components/Number.tsx

```diff
@@ -14,17 +20,42 @@ import {
 import { Box } from '@mui/system';
 import { FormLabel } from '@mui/material';
 
-export const numberTester = rankWith(3, schemaTypeIs('number'));
+export const numberTester = rankWith(
+  5,
+  or(schemaTypeIs('number'), uiTypeIs('Number'))
+);
+
+type NumberOptions = {
+  /*
+  Options only required if the saved data should be converted to a string
+  representation of the number
+  */
+  output?: 'string';
+  /*
+  The output string will be padded with leading 0's up to the number of significant figures specified
+  */
+  sigFigs?: number;
+};
 
 const UIComponent = (props: ControlProps) => {
-  const { data, handleChange, label, path, errors, schema } = props;
-  const [localData, setLocalData] = useState<number | undefined>(data);
+  const { data, handleChange, label, path, errors, schema, uischema } = props;
+  const [localData, setLocalData] = useState<number | undefined>(Number(data));
+
   const onChange = useDebounceCallback(
-    (value: number) => handleChange(path, value),
+    (value: number | string) => handleChange(path, value),
     [path]
   );
   const error = !!errors;
 
+  const options: NumberOptions = uischema?.options ?? {};
+
+  // The selected number can be saved as a string (with optional leading zeroes)
+  // using the "output: string" and "sigFig" options
+  const formatValue = (value: number): number | string => {
+    if (options?.['output'] !== 'string') return value;
+    return String(value).padStart(options?.['sigFigs'] ?? 0, '0');
+  };
```

Comment:
We could do more here (such as rounding to "sigFigs" etc.), but not necessary for now. Can extend this function if/when required.